File: 面试题14-I-剪绳子.py
class Solution:
    def cuttingRope(self, n: int) -> int:
        mylist = list(range(n)) + [0]
        
        for i in range(1, n + 1):
            for j in range(1, i):
                mylist[i] = max(mylist[i], (i - j) * mylist[j])
        
        return mylist[n]

File: 面试题14-I-剪绳子.cpp
class Solution {
public:
    int cuttingRope(int n) {
        int num2;
        
        switch(n % 3){
            case 0: num2 = 0; break;
            case 1: num2 = 2; break;
            case 2: num2 = 1; break;
        }

        switch(n){
            case 2: return 1;
            case 3: return 2;
            default: return pow(2, num2) * pow(3, (n - 2 * num2) / 3);
        } 
    }
};
